include/aos/iam/identhandler.hpp

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright (C) 2024 Renesas Electronics Corporation.
+ * Copyright (C) 2024 EPAM Systems, Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef AOS_IDENTHANDLER_HPP_
+#define AOS_IDENTHANDLER_HPP_
+
+#include "aos/common/types.hpp"
+#include "aos/iam/config.hpp"
+
+namespace aos {
+namespace iam {
+namespace identhandler {
+
+/** @addtogroup iam Identification and Access Manager
+ *  @{
+ */
+
+/**
+ * Subject Changed Observer interface.
+ */
+class SubjectsChangedObserverItf {
+public:
+    /**
+     * Update observer.
+     *
+     * @param[in] updateMessage update message
+     * @returns Error.
+     */
+    virtual Error SubjectsChanged(const Array<StaticString<cSubjectLen>>& messages) = 0;
+
+    /**
+     * Destroys ident observer interface.
+     */
+    virtual ~SubjectsChangedObserverItf() = default;
+};
+
+/**
+ * Ident handler interface.
+ */
+class IdentHandlerItf {
+public:
+    /**
+     * Returns System ID.
+     *
+     * @param[out] systemdID result system ID.
+     * @returns Error.
+     */
+    virtual Error GetSystemID(String& systemdID) = 0;
+
+    /**
+     * Returns unit model.
+     *
+     * @param[out] unitModel result unit model.
+     * @returns Error.
+     */
+    virtual Error GetUnitModel(String& unitModel) = 0;
+
+    /**
+     * Returns subjects.
+     *
+     * @param[out] subjects result subjects.
+     * @returns Error.
+     */
+    virtual Error GetSubjects(Array<StaticString<cSubjectLen>>& subjects) = 0;
+
+    /**
+     * Set subjects changed observer.
+     *
+     * @param[in] observer subjects changed observer.
+     * @returns Error.
+     */
+    virtual Error SetSubjectsChangedObserver(SharedPtr<SubjectsChangedObserverItf> observer) = 0;
```

Comment:
No need for this method. All interface implementers have `Init` method. It should be there.